### ISSUE TITLE
Not balancing UDP stream

### ISSUE THREAD

**Issue by gedOHub** [yyyar/gobetween#128]
Hello,

I was playing around with this software and notices that UDP stream was not balancing

Version: `gobetween v0.5.0`

Config file:

```
#
# gobetween.toml - sample config file
#
# Website: http://gobetween.io
# Documentation: https://github.com/yyyar/gobetween/wiki/Configuration
#


#
# Logging configuration
#
[logging]
level = "info"   # "debug" | "info" | "warn" | "error"
#output = "stdout" # "stdout" | "stderr" | "/path/to/gobetween.log"
output = "/var/log/gobetween/gobetween.log" # "stdout" | "stderr" | "/path/to/gobetween.log"


# REST API server configuration
#
[api]
enabled = true  # true | false
bind = ":8888"  # "host:port"
cors = false    # cross-origin resource sharing

#  [api.basic_auth]   # (optional) Enable HTTP Basic Auth
#  login = "admin"    # HTTP Auth Login
#  password = "1111"  # HTTP Auth Password

#  [api.tls]                        # (optional) Enable HTTPS
#  cert_path = "/path/to/cert.pem"  # Path to certificate
#  key_path = "/path/to/key.pem"    # Path to key


#
# Default values for server configuration, may be overriden in [servers] sections.
# All "duration" fields (for examole, postfixed with '_timeout') have the following format:
# <int><duration> where duration can be one of 'ms', 's', 'm', 'h'.
# Examples: "5s", "1m", "500ms", etc. "0" value means no limit
#
[defaults]
max_connections = 0              # Maximum simultaneous connections to the server
client_idle_timeout = "0"        # Client inactivity duration before forced connection drop
backend_idle_timeout = "0"       # Backend inactivity duration before forced connection drop
backend_connection_timeout = "0" # Backend connection timeout (ignored in udp)



[servers.vflow]
bind = "0.0.0.0:4729"
protocol = "udp"
balance = "leastbandwidth"

  [servers.vflow.udp]
  max_responses = 0
  

  [servers.vflow.discovery]
  kind = "static"
  static_list = [
      "10.24.21.110:4729 weight=3",
      "10.24.21.111:4729 weight=3",
      "10.24.21.112:4729 weight=3"
  ]
  
  [servers.vflow.healthcheck]
  kind = "exec"
  interval = "2s"
  timeout = "2s" 

  exec_command = "/usr/share/exec_healthcheck.sh"
  exec_expected_positive_output = "1"
  exec_expected_negative_output = "0"
```

I have tried all balancing methods listed in [documentation](http://gobetween.io/documentation.html#Balancing)

API output example:
Servers:
```
{
    "max_connections": 0,
    "client_idle_timeout": "0",
    "backend_idle_timeout": "0",
    "backend_connection_timeout": "0",
    "bind": "0.0.0.0:4729",
    "protocol": "udp",
    "balance": "leastbandwidth",
    "sni": null,
    "tls": null,
    "backends_tls": null,
    "udp": {
        "max_requests": 0,
        "max_responses": 0
    },
    "access": null,
    "proxy_protocol": null,
    "discovery": {
        "kind": "static",
        "failpolicy": "keeplast",
        "interval": "0",
        "timeout": "0",
        "static_list": [
            "10.24.21.110:4729 weight=3",
            "10.24.21.111:4729 weight=3",
            "10.24.21.112:4729 weight=3"
        ]
    },
    "healthcheck": {
        "kind": "exec",
        "interval": "2s",
        "passes": 1,
        "fails": 1,
        "timeout": "2s",
        "exec_command": "/usr/share/exec_healthcheck.sh",
        "exec_expected_positive_output": "1",
        "exec_expected_negative_output": "0"
    }
}
```

Stats:
```
{
    "active_connections": 0,
    "rx_total": 0,
    "tx_total": 1122560856,
    "rx_second": 0,
    "tx_second": 1845348,
    "backends": [
        {
            "host": "10.24.21.110",
            "port": "4729",
            "priority": 1,
            "weight": 3,
            "stats": {
                "live": true,
                "total_connections": 0,
                "active_connections": 0,
                "refused_connections": 0,
                "rx": 0,
                "tx": 1118870160,
                "rx_second": 0,
                "tx_second": 1862184
            }
        },
        {
            "host": "10.24.21.111",
            "port": "4729",
            "priority": 1,
            "weight": 3,
            "stats": {
                "live": true,
                "total_connections": 0,
                "active_connections": 0,
                "refused_connections": 0,
                "rx": 0,
                "tx": 0,
                "rx_second": 0,
                "tx_second": 0
            }
        },
        {
            "host": "10.24.21.112",
            "port": "4729",
            "priority": 1,
            "weight": 3,
            "stats": {
                "live": true,
                "total_connections": 0,
                "active_connections": 0,
                "refused_connections": 0,
                "rx": 0,
                "tx": 0,
                "rx_second": 0,
                "tx_second": 0
            }
        }
    ]
}
```

Any help on this?

**Comment by illarion** [comment]
UDP is sessionless protocol, working on level of individual packets, but in order to support multiple packets exchange, gobetween associates elected backend with client address. This association is dropped if client_idle_timeout or backend_idle_timeout is overdue, or if there were more than max_responses responses from backend, (or max_requests from the client). Default value of 0 means no such limit. It seems you are using some application sending udp packets, so that gb assumes this app needs to talk to one server, not to all of them.

TLDR: change max_responses to 1 or whatever value is needed for you application. If it helps, I'll close the issue.

**Comment by gedOHub** [comment]
@illarion Problem is that my backed application do not respond. It consumes only one way communication. Can gobetween be quite dumb and just send next packet to next backed?

**Comment by illarion** [comment]
Ok, than set max_responses to 0 and max_requests to 1, so that link will disappear after one packet and the next one will go to another backend

**Comment by illarion** [comment]
@gedOHub did it help?

**Comment by gedOHub** [comment]
@illarion Yes. It helped. Thank you

**Comment by Joshigangadharan** [comment]
@gedOHub can you please share an example for writing  "/usr/share/exec_healthcheck.sh" healthcheck file.  I am looking for a template to do healthcheck for DNS (UDP). Thanks much in advance.
